### PR TITLE
Update Documentation content

### PR DESCRIPTION
**Pull request by karlchillmaid**
This PR updates the Documentation page to:
- make the target user and purpose of the page clearer
- tell users how long it might take to integrate with the API
- tell users they can integrate directly with the API
- provide important information about integrating directly with the API

### CODE REVIEW

**Comment by karlchillmaid** [comment]
Updates made after discussing with @servingUpAces and @jonathanglassman.

**Comment by yahoopete** [comment]
I think we need to say that we recommend using one of our clients here. This will invite requests for the API spec I think...  Also think saying a full integration takes a couple of days is a bit misleading for teams who aren't just changing supplier, because most of the work is customising their systems as a result of the new functionality. Whether that's worth stating or not is another story.....

**Comment by karlchillmaid** [comment]
@yahoopete 
Content updated based on your comments.

**Comment by quis** [comment]
In each of the client documentation we say:

> We recommend that you use this client library rather than use the [GOV.UK Notify API](https://github.com/alphagov/notifications-api) directly, as there is no documentation for using the API in this way.

I’d rather we just said it here, rather than there, now that we’re explaining it better here.

File: app/templates/views/documentation.html [hunk 1]
```diff
@@ -22,5 +22,13 @@ <h2 class="heading-medium">Clients</h2>
...
     {% endfor %}
   </ul>
-
+    <p>A developer should be able to set up the API client and start sending test messages in around 30 minutes. A full integration can take a few days, depending on the other systems you're using.</p>
+  <h2>Integrating directly with the API</h2>
+    <p>We recommend using our client documentation rather than integrating directly with the API.</p>
+    <p>There's no documentation for using the API in this way. You'll still need to read the client documentation to understand:</p>
+  <ul class="list list-bullet">
+    <li>what the API client does</li>
+    <li>what development work you'll need to do</li>
```
Comment (quis) [inline app/templates/views/documentation.html]:
```suggestion
    <li>what development work you’ll need to do</li>
```

File: app/templates/views/documentation.html [hunk 2]
```diff
@@ -22,5 +22,13 @@ <h2 class="heading-medium">Clients</h2>
...
+  </ul>
+  <p>To send a message you'll need to create an HTTPS request and add an authorisation header with your API key. The API key is encoded with JSON Web Tokens (JWT).</p>
```
Comment (quis) [inline app/templates/views/documentation.html]:
```suggestion
  <p>To send a message you’ll need to create an HTTPS request and add an authorisation header with your API key. The API key is encoded with JSON Web Tokens (JWT).</p>
```

File: app/templates/views/documentation.html [hunk 3]
```diff
@@ -22,5 +22,13 @@ <h2 class="heading-medium">Clients</h2>
       <li><a href="https://docs.notifications.service.gov.uk/{{ key }}.html" target="_blank" rel="noopener">{{ label }}</a></li>
     {% endfor %}
   </ul>
-
+    <p>A developer should be able to set up the API client and start sending test messages in around 30 minutes. A full integration can take a few days, depending on the other systems you're using.</p>
+  <h2>Integrating directly with the API</h2>
+    <p>We recommend using our client documentation rather than integrating directly with the API.</p>
+    <p>There's no documentation for using the API in this way. You'll still need to read the client documentation to understand:</p>
```
Comment (quis) [inline app/templates/views/documentation.html]:
```suggestion
    <p>There’s no documentation for using the API in this way. You'll still need to read the client documentation to understand:</p>
```

File: app/templates/views/documentation.html [hunk 4]
```diff
@@ -22,5 +22,13 @@ <h2 class="heading-medium">Clients</h2>
       <li><a href="https://docs.notifications.service.gov.uk/{{ key }}.html" target="_blank" rel="noopener">{{ label }}</a></li>
     {% endfor %}
   </ul>
-
+    <p>A developer should be able to set up the API client and start sending test messages in around 30 minutes. A full integration can take a few days, depending on the other systems you're using.</p>
```
Comment (quis) [inline app/templates/views/documentation.html]:
```suggestion
    <p>A developer should be able to set up the API client and start sending test messages in around 30 minutes. A full integration can take a few days, depending on the other systems you’re using.</p>
```

**Comment by karlchillmaid** [comment]
@quis 
> I’d rather we just said it here, rather than there, now that we’re explaining it better here.

I'll get PRs raised for each of the clients

**Pull request merged**